libyul/AsmParser.cpp

```diff
@@ -134,84 +134,97 @@ void Parser::fetchSourceLocationFromComment()
 		R"~~((?:^|\s+)(@[a-zA-Z0-9\-_]+)(?:\s+|$))~~", // tag, e.g: @src
 		regex_constants::ECMAScript | regex_constants::optimize
 	);
-	static regex const srcTagArgsRegex = regex(
-		R"~~(^(-1|\d+):(-1|\d+):(-1|\d+)(?:\s+|$))~~"  // index and location, e.g.: 1:234:-1
-		R"~~(("(?:[^"\\]|\\.)*"?)?)~~",                // optional code snippet, e.g.: "string memory s = \"abc\";..."
-		regex_constants::ECMAScript | regex_constants::optimize
-	);
 
-	string const commentLiteral = m_scanner->currentCommentLiteral();
+	string_view commentLiteral = m_scanner->currentCommentLiteral();
 	SourceLocation const commentLocation = m_scanner->currentCommentLocation();
-	smatch tagMatch;
-	string::const_iterator position = commentLiteral.begin();
+	match_results<string_view::const_iterator> tagMatch;
 
-	while (regex_search(position, commentLiteral.end(), tagMatch, tagRegex))
+	langutil::SourceLocation sourceLocation = m_debugDataOverride->location;
+
+	while (regex_search(commentLiteral.cbegin(), commentLiteral.cend(), tagMatch, tagRegex))
 	{
 		solAssert(tagMatch.size() == 2, "");
-		position += tagMatch.position() + tagMatch.length();
+		commentLiteral = commentLiteral.substr(static_cast<size_t>(tagMatch.position() + tagMatch.length()));
 
 		if (tagMatch[1] == "@src")
 		{
-			smatch srcTagArgsMatch;
-			if (!regex_search(position, commentLiteral.end(), srcTagArgsMatch, srcTagArgsRegex))
-			{
-				m_errorReporter.syntaxError(
-					8387_error,
-					commentLocation,
-					"Invalid values in source location mapping. Could not parse location specification."
-				);
-
-				// If the arguments to @src are malformed, we don't know where they end so we can't continue.
-				return;
-			}
-
-			solAssert(srcTagArgsMatch.size() == 5, "");
-			position += srcTagArgsMatch.position() + srcTagArgsMatch.length();
-
-			if (srcTagArgsMatch[4].matched && (
-				!boost::algorithm::ends_with(srcTagArgsMatch[4].str(), "\"") ||
-				boost::algorithm::ends_with(srcTagArgsMatch[4].str(), "\\\"")
-			))
-			{
-				m_errorReporter.syntaxError(
-					1544_error,
-					commentLocation,
-					"Invalid code snippet in source location mapping. Quote is not terminated."
-				);
-				return;
```

Comment:
Ah, this is a small difference: Previously, this fully aborted the parsing, while the other cases did not abort. I think it's fine to abort in all cases.

Reply:
Ok, one could also argue in the other direction and not abort so that we at least get valid data for the other tags. I'll change it.

Reply:
You code seems equivalent. It does not even break this test: https://github.com/ethereum/solidity/blob/febbd8deb31e3dafafc72e2e4704cd8b804cc46c/test/libyul/Parser.cpp#L761-L776

You technically do not abort the loop but you still return an empty source location and an empty string view so the loop terminates.